File: JavaRxSample/app/src/main/java/com/coderbunker/javarxsample/MainActivity.java
package com.coderbunker.javarxsample;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.coderbunker.javarxsample.citylist.ListActivity;
import com.coderbunker.javarxsample.operators.OperatorsActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_operators);

        Button showList = (Button) findViewById(R.id.show_list);
        showList.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d(App.TAG, "Show list");
                ListActivity.start(v.getContext());
            }
        });

        Button showOperators = (Button) findViewById(R.id.show_operators);
        showOperators.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d(App.TAG, "Show operators");
                OperatorsActivity.start(v.getContext());
            }
        });
    }
}

File: app/src/main/java/com/harman/motionlayoutanimation/MainActivity.kt
package com.harman.motionlayoutanimation

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.constraint.motion.MotionLayout
import android.view.View
import android.widget.SeekBar
import kotlin.math.ceil

class MainActivity : AppCompatActivity() {

    var motion_container: MotionLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        motion_container = findViewById<MotionLayout>(R.id.motion_container)
        motion_container!!.setTransitionListener(
                object: MotionLayout.TransitionListener {
                    override fun onTransitionChange(motionLayout: MotionLayout?,
                                                    startId: Int,
                                                    endId: Int,
                                                    progress: Float) {
                        var seekbar = findViewById<SeekBar>(R.id.seekbar)
                        seekbar.progress = ceil(progress * 100).toInt()
                    }

                    override fun onTransitionCompleted(motionLayout: MotionLayout?,
                                                       currentId: Int) {
                        if(currentId == R.id.ending_set) {
                            // Return to original constraint set
                            motion_container!!.transitionToStart()
                        }
                    }
                }
        )
    }

    fun start(v: View) {
        motion_container!!.transitionToEnd()
    }

}
